plotly/plotly/plotly.py

```diff
@@ -1216,6 +1218,31 @@ def validate_credentials(credentials):
         raise exceptions.PlotlyLocalCredentialsError()
 
 
+def secret_key(r, **payload):
+    """
+    update plot's url to include the scret key
+
+    """
+
+    plot_id = r['url'].replace("https://plot.ly/~" + payload['un'] + "/", "")
+    url = "https://api.plot.ly/v2/files/" + payload['un'] + ":" + plot_id
+    plot_kwargs = json.loads(payload['kwargs'])
+    response = requests.patch(url,
+                              headers={'Plotly-Client-Platform': 'Python'},
+                              auth=HTTPBasicAuth(payload['un'],
+                                                 payload['key']),
+                              data={"share_key_enabled":
+                                    plot_kwargs['share_key_enabled'],
+                                    "world_readable":
+                                    plot_kwargs['world_readable'],
+                                    "filename": plot_kwargs['filename'],
+                                    "fileopt": plot_kwargs['fileopt']})
+    response = json.loads(response.content)
```

Comment:
:cow2: It's helpful to not change the value type of a variable. So earlier `response` was a `Response` object, and now it's a `dict`. --> `response_data = json.loads(response.content)`?